controllers/status/status.go

```diff
@@ -210,7 +216,135 @@ func RemoveComponentCondition(conditions *[]conditionsv1.Condition, component st
+	Version version.OperatorVersion `json:"version,omitempty"`
+	RepoURL string                  `json:"repoURL,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type ComponentStatus struct {
+	Releases []ComponentReleaseStatus `json:"releases,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type CodeFlareStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type DashboardStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type WorkbenchesStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type ModelMeshServingStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type DataSciencePipelinesStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type KserveStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type KueueStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type RayStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type TrustyAIStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
 type ModelRegistryStatus struct {
 	RegistriesNamespace string `json:"registriesNamespace,omitempty"`
+	ComponentStatus     `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type TrainingOperatorStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type ComponentsStatus struct {
+	CodeFlare            *CodeFlareStatus            `json:"codeflare,omitempty"`
+	Dashboard            *DashboardStatus            `json:"dashboard,omitempty"`
+	DataSciencePipelines *DataSciencePipelinesStatus `json:"datasciencepipelines,omitempty"`
+	ModelMeshServing     *ModelMeshServingStatus     `json:"modelmeshserving,omitempty"`
+	ModelRegistry        *ModelRegistryStatus        `json:"modelregistry,omitempty"`
+	Kserve               *KserveStatus               `json:"kserve,omitempty"`
+	Kueue                *KueueStatus                `json:"kueue,omitempty"`
+	Ray                  *RayStatus                  `json:"ray,omitempty"`
+	TrustyAI             *TrustyAIStatus             `json:"trustyai,omitempty"`
+	TrainingOperator     *TrainingOperatorStatus     `json:"trainingoperator,omitempty"`
+	Workbenches          *WorkbenchesStatus          `json:"workbenches,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type ReleaseFileMeta struct {
+	Releases []ComponentReleaseStatusMeta `json:"releases,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type ComponentReleaseStatusMeta struct {
+	Name    string `yaml:"name,omitempty"`
+	Version string `yaml:"version,omitempty"`
+	RepoURL string `yaml:"repoURL,omitempty"`
+}
+
+// GetReleaseVersion reads odh_metadata.yaml file and parses release information.
+// If version is not set or set to "", return empty {}.
+func GetReleaseVersion(defaultManifestPath string, componentName string) ComponentStatus {
```

Comment:
I think `GetReleaseVersion` should return a `[]ComponentReleaseStatus` not the entire `ComponentStatus`

Reply:
@lburgazzoli but incase of yaml file not present in the components repo, we discussed to show empty struct against component names. such error conditions are handled in the functions itself so as returning empty struct. When returning `[]ComponentReleaseStatus` , will have to handle error cases in components.

Reply:
you can return an empty list, if you sue `omitempty` in the tags, then it won't not show up

Reply:
@lburgazzoli yep made this changes now. :)

Reply:
> I think `GetReleaseVersion` should return a `[]ComponentReleaseStatus` not the entire `ComponentStatus`

If it returns ReleaseStatus, why it's still called GetReleaseVersion? (asked @Sara4994 already)

Reply:
@ykaliuta renamed it now. thanks